msgpack/_unpacker.pyx

```diff
@@ -129,12 +131,27 @@ def unpackb(object packed, object object_hook=None, object list_hook=None,
     cdef Py_ssize_t off = 0
     cdef int ret
 
-    cdef char* buf
+    cdef Py_buffer view
+    cdef char* buf = NULL
     cdef Py_ssize_t buf_len
     cdef char* cenc = NULL
     cdef char* cerr = NULL
-
-    PyObject_AsReadBuffer(packed, <const void**>&buf, &buf_len)
+    cdef char buffer_supported = 0
+
+    if PyObject_CheckBuffer(packed):
+        buffer_supported = 1
+        if PyObject_GetBuffer(packed, &view, PyBUF_SIMPLE) == 0:
+            if view.itemsize != 1:
+                PyBuffer_Release(&view)
+                raise ValueError("cannot unpack from multi-byte object")
+            buf_len = view.len
+            buf = <char*> view.buf
+    else:
+        PyObject_AsReadBuffer(packed, <const void**>&buf, &buf_len)
+        PyErr_WarnEx(DeprecationWarning,
+                     "unpacking %s requires old buffer protocol,"
+                     "which will be removed in msgpack 1.0" % type(packed),
```

Comment:
Error messages doesn't start with upper case.  It means it isn't whole sentence.
FWIW, in Golang:

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Print("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

I grepped CPython's Lib directry and I feel Python has same convention.